client/post-editor/index.js

```diff
@@ -13,18 +13,18 @@ import config from 'config';
 module.exports = function() {
 	page( '/post', controller.pressThis, sitesController.siteSelection, sitesController.sites );
 	page( '/post/new', () => page.redirect( '/post' ) ); // redirect from beep-beep-boop
-	page( '/post/:site?/:post?', sitesController.siteSelection, sitesController.fetchJetpackSettings, controller.post );
```

Comment:
There might still be components that rely on site settings being loaded properly. Currently removing this will cause a subtle regression with the default post format being fetched from the `site` but not from the site's `settings` (for Jetpack sites). 

This regression will be no more when we merge #13329, which retrieves the site default post format from Redux (we already have a `<QuerySiteSettings />` in the editor drawer). There is a good discussion about this case in the PR.